google/cloud/spanner/samples/samples.cc

```diff
@@ -476,31 +396,41 @@ void QueryDataWithStruct(google::cloud::spanner::Client client) {
 }
 //! [END spanner_query_data_with_struct]
 
-void QueryDataWithStructCommand(std::vector<std::string> const& argv) {
-  if (argv.size() != 3) {
-    throw std::runtime_error(
-        "query-data-with-struct <project-id> <instance-id> <database-id>");
-  }
-  QueryDataWithStruct(MakeSampleClient(argv[0], argv[1], argv[2]));
-}
-
 int RunOneCommand(std::vector<std::string> argv) {
   using CommandType = std::function<void(std::vector<std::string> const&)>;
 
-  std::map<std::string, CommandType> commands = {
+  using SampleFunction = void (*)(google::cloud::spanner::Client);
+
+  using CommandMap = std::map<std::string, CommandType>;
+  auto make_command_entry = [](char const* sample_name, SampleFunction sample) {
+    auto make_command = [](char const* sample_name, SampleFunction sample) {
+      return [sample_name, sample](std::vector<std::string> const& argv) {
+        if (argv.size() != 3) {
+          throw std::runtime_error(std::string(sample_name) +
+                                   " <project-id> <instance-id> <database-id>");
+        }
+        sample(MakeSampleClient(argv[0], argv[1], argv[2]));
+      };
+    };
+    return CommandMap::value_type(sample_name,
+                                  make_command(sample_name, sample));
+  };
+
+  CommandMap commands = {
       {"create-database", &CreateDatabase},
       {"add-column", &AddColumn},
       {"drop-database", &DropDatabase},
```

Comment:
So I'm clear, these first three are different because they connect to the "admin" interface and don't use a `spanner::Client`?

Reply:
Yes, that is why.